### ISSUE TITLE
cannot import name 'Buffer' from 'typing_extensions' (/usr/local/lib/python3.10/dist-packages/typing_extensions.py)

### ISSUE THREAD

**Issue by Balogunhabeeb14** [ydataai/ydata-profiling#1509]
### Current Behaviour

 I get this error:
cannot import name 'Buffer' from 'typing_extensions' (/usr/local/lib/python3.10/dist-packages/typing_extensions.py)

when I try importing from ydata_profiling import ProfileReport

### Expected Behaviour

I should get a Report as shown on the https://docs.profiling.ydata.ai/4.6/

### Data Description

it is the public diabetes.csv public available dataset

### Code that reproduces the bug

```Python
from ydata_profiling import ProfileReport
```

### pandas-profiling version

pandas-profiling==3.6.6

### Dependencies

```Text
numpy==1.23.5
pandas==1.5.3
```

### OS

Macos

### Checklist

- [X] There is not yet another bug report for this issue in the [issue tracker](https://github.com/ydataai/pandas-profiling/issues)
- [X] The problem is reproducible from this bug report. [This guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) can help to craft a minimal bug report.
- [X] The issue has not been resolved by the entries listed under [Common Issues](https://pandas-profiling.ydata.ai/docs/master/pages/support_contrib/common_issues.html).

**Comment by KhethiweDlamini** [comment]
Hi, you've probably gotten a solution by now. Nonetheless, I fixed a similar issue by updating my Python from version 3.10 to version 3.12 as the 'Buffer' extension is available on the latest version. Everything worked after that. 😁

**Comment by fabclmnt** [comment]
Hi @KhethiweDlamini ,

thank you for your suggestion, have in mind that ydata-profiling distribution is not available for python 3.12. Not sure how have you installed the package, but it is not officially supported due to conflicts.

**Comment by Balogunhabeeb14** [comment]
Many thanks for the suggestions @KhethiweDlamini  I tried changing the version of python, however that didn't work.

@fabclmnt to answer your question, It was on google colab, though I don't think that should be a problem.

Anyways, this recent version which you shared worked just now.

Thank you. It may be worth it if you can update the documentations.
